eparticlectr/eparticlectr.cpp

```diff
@@ -543,15 +788,156 @@ void eparticlectr::logpropinfo( uint64_t proposal_id, name proposer, uint64_t wi
     require_auth( _self );
 }
 
+[[eosio::action]]
+void eparticlectr::logpropinfex( uint64_t proposal_id, name proposer, uint64_t wiki_id, std::string slug, ipfshash_t ipfs_hash, std::string lang_code, uint64_t group_id, std::string comment, std::string memo, uint32_t starttime, uint32_t endtime, std::string proxied_for, std::string extra_note) {
+    require_auth( _self );
+}
+
 [[eosio::action]]
 void eparticlectr::curatelist( name user, std::string title, std::string description, std::vector<std::string> wikis, bool ranked ) {
     require_auth(user);
     check(title.size() < 100, "Title is too long. Max 100 chars");
     check(description.size() < 300, "Description is too long. Max 300 chars");
 }
 
-//void eparticlectr::logboostinv( uint64_t boost_id, name booster, std::string slug, std::string lang_code, uint64_t amount, std::string memo, uint32_t timestamp) {
-//    require_auth( _self );
-//}
 
-EOSIO_DISPATCH( eparticlectr, (brainclmid)(slashnotify)(finalize)(oldvotepurge)(propose2)(rewardclmid)(vote)(logpropres)(logpropinfo)(mkreferendum)(curatelist) )
+[[eosio::action]]
+void eparticlectr::migratestkes( uint32_t loop_limit ) {
+    require_auth( name("evrpdcronjob") );
+
+    // Initialize the two tables 
+    staketbl stakestable( _self, _self.value );
+    staketblex stakestable_extra( _self, _self.value );
+
+    // Loop through the old proposals 
+    uint32_t counter = 0;
+    for ( auto stakes_it = stakestable.begin(); stakes_it != stakestable.end() && counter < loop_limit; stakes_it++ ) {
+        // Copy the old reward over to the new format
+        stakestable_extra.emplace( _self, [&]( auto& s ) {
+            s.id = stakes_it->id;
+            s.user = stakes_it->user;
+            s.amount = stakes_it->amount;
+            s.timestamp = stakes_it->timestamp;
+            s.completion_time = stakes_it->completion_time;
+            s.proxied_for = "";
+            s.extra_note = "";
+        });
+
+        // Delete the old stakes
+        // stakestable.erase(stakes_it);
```

Comment:
if you dont erase, the loop_limit won't work. You need to cleanup rows in order to be able to run multiple times the command (apply this comment for all the other migrations)